libs/full/segmented_algorithms/include/hpx/parallel/segmented_algorithms/detail/dispatch.hpp

```diff
@@ -185,30 +143,6 @@ namespace hpx { namespace parallel { inline namespace v1 { namespace detail {
         }
     };
 
-    template <typename Iterator1, typename Iterator2>
-    struct algorithm_result_helper<future<std::pair<Iterator1, Iterator2>>,
-        std::enable_if_t<
-            hpx::traits::is_segmented_local_iterator_v<Iterator1> ||
-            hpx::traits::is_segmented_local_iterator_v<Iterator2>>>
-    {
-        using traits1 = hpx::traits::segmented_local_iterator_traits<Iterator1>;
-        using traits2 = hpx::traits::segmented_local_iterator_traits<Iterator2>;
-
-        using arg_type = std::pair<typename traits1::local_raw_iterator,
-            typename traits2::local_raw_iterator>;
-        using result_type = std::pair<typename traits1::local_iterator,
-            typename traits2::local_iterator>;
-
-        HPX_FORCEINLINE static future<result_type> call(future<arg_type>&& f)
-        {
-            return hpx::make_future<result_type>(
-                HPX_MOVE(f), [](arg_type&& p) -> result_type {
-                    return std::make_pair(
-                        traits1::remote(p.first), traits2::remote(p.second));
-                });
-        }
-    };
-
```

Comment:
The specialization for `future<hpx::tuple<Iterator1, Iterator2, Iterator3>>` below might not be needed anymore either.